### PR TITLE
[FG-5091] Consolidate exporter CLI and UI into monorepo

### PR DESCRIPTION
**Pull request by neosh11**
## Summary

Consolidates the [exporter UI](https://github.com/SafetyCulture/safetyculture-exporter-ui/) into this repository alongside the CLI, creating a single monorepo with a unified release process.

## Problem

The UI frequently falls out of sync with CLI changes because they live in separate repositories. Releasing requires coordinating across two repos and waiting for CLI artifacts to publish before the UI can build against them.

## What changed

**Repository structure** — moved CLI source into `cli/` and UI source into `ui/`:
```
├── cli/          # Go CLI (module path unchanged)
├── ui/           # Wails v2 desktop app (React + Tailwind + shadcn/ui)
│   └── frontend/ # React frontend (pnpm)
├── go.work       # Go workspace linking both modules
└── .github/      # Restructured CI workflows
```

**Go workspace** — `go.work` links `cli/` and `ui/` so the UI always builds against the latest CLI code locally, no published artifacts required.

**CI/CD restructured** into three workflows:
- `ci-cli.yml` — CLI unit tests, integration tests (Postgres/MySQL/SQL Server), linting, govulncheck. Runs with `GOWORK=off` for module isolation.
- `ci-ui.yml` — UI Go tests, Wails build smoke test, govulncheck.
- `release.yml` — Cross-platform builds for CLI (linux/darwin/windows) and UI (macOS universal with notarization, Windows with code signing). Single GitHub Release with all artifacts.

**Other changes:**
- Updated README for monorepo development workflow
- Added `.yamllint.yml` config for CI linting
- Bumped Go toolchain to 1.24.4 (resolves stdlib vulnerabilities)
- Removed stale `package-lock.json` (project uses pnpm)

## Key design decisions

- **CLI module path preserved** (`github.com/SafetyCulture/safetyculture-exporter`) — external consumers are unaffected.
- **`GOWORK=off` for CLI CI** — ensures the CLI's `go.mod` stays authoritative and doesn't pull in Wails/CGO deps.
- **Separate Go modules** — keeps CGO (Wails) deps isolated from the pure-Go CLI.

## Checklist

- [X] Is this PR a reasonable size?

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: .github/workflows/ci-cli.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'cli/**'
+      - 'go.work'
+      - 'go.work.sum'
+      - '.github/workflows/ci-cli.yml'
+      - '.pre-commit-config.yaml'
+
+jobs:
```
[the remark this answers is not in the document]
Comment (neosh11) [reply .github/workflows/ci-cli.yml:14]: This is a public repo. The code is just refactoring the existing workflows.

**Review by HarryTranAU — APPROVED** [review]
